### PR TITLE
Update convict methods to use templete literal types

### PR DESCRIPTION
**Pull request by msuntharesan**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/mozilla/node-convict/blob/master/CHANGELOG.md>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/convict/index.d.ts [hunk 1]
```diff
@@ -200,14 +187,26 @@ declare namespace convict {
...
+        getEnv(): string[];
+
+        /**
+         * Gets the array of process arguments, using the override passed to the
+         * convict function or process.argv if no override was passed.
+         */
+        getArgs(): string[];
     }
-}
+  }
```
Comment (louis79719) [inline types/convict/index.d.ts:203]: I believe this indent is incorrect.

File: types/convict/index.d.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ declare namespace convict {
     interface ValidateOptions {
         /**
          * If set to warn, any properties specified in config files that are not declared in
-         * the schema will print a warning. This is the default behavior. If set to strict,
+         * the schema will print a warning. This is the dƒefault behavior. If set to strict,
```
Comment (louis79719) [inline types/convict/index.d.ts:23]: typo?

**Review by louis79719 — CHANGES_REQUESTED** [review]

File: types/convict/index.d.ts (hunk 1 above)
Comment (msuntharesan) [reply types/convict/index.d.ts:203]: Thanks. it is corrected.

File: types/convict/index.d.ts (hunk 2 above)
Comment (msuntharesan) [reply types/convict/index.d.ts]: My shift key isn't working 😢. Thanks for catching it.

File: types/convict/convict-tests.ts [hunk 3]
```diff
@@ -139,26 +137,28 @@ if (conf.has('key')) {
...
 
 const schemaVal = conf.getSchema().properties.db.properties.port.default;
 
 conf.get();
+// @ts-expect-error Trying to access no existing property
 conf.get('unknownkey');
 conf.get('db');
 conf.get('db.ip');
-conf.get<'db', 'ip'>('db.ip');
+conf.get('db.ip');
```
Comment (louis79719) [inline types/convict/convict-tests.ts:158]: If you remove the <...> syntax, then this line is duplicated with the above one. Is this needed?

Comment (msuntharesan) [reply types/convict/convict-tests.ts:158]: Yes. they are duplicates. I'll have it removed

**Review by louis79719 — APPROVED** [review]

**Comment by msuntharesan** [comment]
Ready to merge

**Pull request merged**